src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketTeamMetadataAction.java

```diff
@@ -54,27 +54,27 @@
 
     private final BitbucketAvatarCacheSource avatarSource;
 
-    public BitbucketTeamMetadataAction(String serverUrl, StandardCredentials credentials, String team) {
-        avatarSource = new BitbucketAvatarCacheSource(serverUrl, credentials, team);
+    public BitbucketTeamMetadataAction(String serverUrl, BitbucketCredentials bbCredentials, String team) {
+        avatarSource = new BitbucketAvatarCacheSource(serverUrl, bbCredentials, team);
     }
 
     public static class BitbucketAvatarCacheSource implements AvatarCacheSource, Serializable {
         private static final long serialVersionUID = 1L;
         private final String serverUrl;
-        private final StandardCredentials credentials;
+        private final BitbucketCredentials bbCredentials;
```

Comment:
If we change the types here, we might need to have some migration code as those are persisted in `state.xml` of the Team folder. I am not sure if the actions will be properly loaded when upgrading. See also https://github.com/jenkinsci/branch-api-plugin/blob/branch-api-2.6.4/src/main/java/jenkins/branch/OrganizationFolder.java#L1635-L1708. 

Also, the fields `serverUrl` and `repoOwner` are therefore redundant here. 

Maybe we can record the SCMNavigator ID and have a transient field to look it up and hold it in memory. Or just use the item full name to look up the context that we need to pass to Bitbucket Credentials.